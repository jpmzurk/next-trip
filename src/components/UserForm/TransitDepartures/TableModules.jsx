import React from "react";
import {
  TableRow,
  TableHead,
  TableFooter,
  TableBody,
  IconButton,
  CircularProgress,
  TableCell,
} from "@mui/material/";
import "../FadeIn.css";
import "./Departures.css";
import AddCircleOutlineOutlinedIcon from "@mui/icons-material/AddCircleOutlineOutlined";
import RemoveCircleOutlineOutlinedIcon from "@mui/icons-material/RemoveCircleOutlineOutlined";

const headerStyles = {
  fontWeight: 700,
  color: "#626462",
  lineHeight: 1.1,
};
const subHeaderStyles = {
  fontWeight: 700,
  color: "#626462",
  fontSize: "1rem",
  letterSpacing: ".5px",
};

const bodyStyles = {
  color: "#626462",
  fontWeight: 500,
  fontSize: "1.2rem",
};

const DeparturesHeader = ({ __stopId }) => {
  return (
    <TableHead className="fadeIn">
      <TableRow>
        <TableCell
          sx={{ fontSize: "1.8rem", width: "80%", pl: 3, ...headerStyles }}
          colSpan={2}
        >
          {__stopId.StopLabel}
        </TableCell>
        <TableCell
          sx={{ fontSize: "1rem", pr: 3, ...headerStyles }}
          align="right"
        >
          <strong>Stop #</strong>: {__stopId.StopID}
        </TableCell>
      </TableRow>
      <TableRow className="tableSubheader">
        <TableCell sx={{ pl: 3, ...subHeaderStyles }}> Route </TableCell>
        <TableCell sx={{ ...subHeaderStyles }}>Destination</TableCell>
        <TableCell align="right" sx={{ pr: 3, ...subHeaderStyles }}>
          Departs
        </TableCell>
      </TableRow>
    </TableHead>
  );
};

function DeparturesBody({ departures }) {
  return (
    <TableBody className="fadeIn">
      {departures.map((departure) => (
        <TableRow key={departure.DepartureTime}>
          <TableCell name="route" sx={{ pl: 3, ...bodyStyles }}>
            {departure.Route}
          </TableCell>
          <TableCell name="stop" sx={{ width: 500, ...bodyStyles }}>
            {departure.Description}
          </TableCell>
          <TableCell
            name="departures"
            sx={{ pr: 3, ...bodyStyles }}
            align="right"
          >
            {departure.DepartureText}
          </TableCell>
        </TableRow>
      ))}
    </TableBody>
  );
}

const DeparturesFooter = ({ handleExpand, isExpanded }) => {
  return (
    <TableFooter className="fadeIn">
      <TableRow>
        <TableCell>
          <IconButton
            name="iconButton"
            aria-label="open or close"
            onClick={handleExpand}
          >
            {isExpanded ? (
              <>
                <RemoveCircleOutlineOutlinedIcon
                  aria-label="close"
                  name="collapse"
                  color="primary"
                />
              </>
            ) : (
              <>
                <AddCircleOutlineOutlinedIcon
                  aria-label="open"
                  name="expand"
                  color="primary"
                />
              </>
            )}
          </IconButton>
        </TableCell>
      </TableRow>
    </TableFooter>
  );
};

const NoDepartures = () => {
  return (
    <TableBody className="fadeIn">
      <TableRow>
        <TableCell></TableCell>
        <TableCell>There are no further departures at this time</TableCell>
        <TableCell></TableCell>
      </TableRow>
    </TableBody>
  );
};

const Loading = () => {
  return (
    <TableBody className="fadeIn">
      <TableRow>
        <TableCell></TableCell>
        <TableCell>
          <CircularProgress />
        </TableCell>
        <TableCell></TableCell>
      </TableRow>
    </TableBody>
  );
};

export {
  DeparturesHeader,
  DeparturesBody,
  DeparturesFooter,
  NoDepartures,
  Loading,
};
