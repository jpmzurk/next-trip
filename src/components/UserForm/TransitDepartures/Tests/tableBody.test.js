import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { DeparturesBody } from "../TableModules";
import "../../../../enzymeConfig";

describe("TableBody should render a mapped list", () => {
  const props = {
    departures: [
      { Route: "3", Description: "Rosedale/Rapid", DepartureText: "3 mins" },
      { Route: "3", Description: "Rosedale/Rapid", DepartureText: "12 mins" },
      { Route: "3", Description: "Rosedale/Rapid", DepartureText: "6:44" },
      { Route: "3", Description: "Rosedale/Rapid", DepartureText: "6:56" },
    ],
  };

  it("should match snapshot", () => {
    const tableBody = shallow(<DeparturesBody {...props} />);
    expect(shallowToJson(tableBody)).toMatchSnapshot();
  });
});
