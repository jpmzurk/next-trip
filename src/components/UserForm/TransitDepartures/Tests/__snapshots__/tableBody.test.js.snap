// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody should render a mapped list should match snapshot 1`] = `
<ForwardRef(TableBody)
  className="fadeIn"
>
  <ForwardRef(TableRow)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pl": 3,
        }
      }
    >
      3
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "width": 500,
        }
      }
    >
      Rosedale/Rapid
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      align="right"
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pr": 3,
        }
      }
    >
      3 mins
    </ForwardRef(TableCell)>
  </ForwardRef(TableRow)>
  <ForwardRef(TableRow)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pl": 3,
        }
      }
    >
      3
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "width": 500,
        }
      }
    >
      Rosedale/Rapid
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      align="right"
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pr": 3,
        }
      }
    >
      12 mins
    </ForwardRef(TableCell)>
  </ForwardRef(TableRow)>
  <ForwardRef(TableRow)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pl": 3,
        }
      }
    >
      3
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "width": 500,
        }
      }
    >
      Rosedale/Rapid
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      align="right"
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pr": 3,
        }
      }
    >
      6:44
    </ForwardRef(TableCell)>
  </ForwardRef(TableRow)>
  <ForwardRef(TableRow)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pl": 3,
        }
      }
    >
      3
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "width": 500,
        }
      }
    >
      Rosedale/Rapid
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      align="right"
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pr": 3,
        }
      }
    >
      6:56
    </ForwardRef(TableCell)>
  </ForwardRef(TableRow)>
  <ForwardRef(TableRow)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pl": 3,
        }
      }
    >
      3
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "width": 500,
        }
      }
    >
      Rosedale/Rapid
    </ForwardRef(TableCell)>
    <ForwardRef(TableCell)
      align="right"
      sx={
        Object {
          "color": "#626462",
          "fontSize": "1.2rem",
          "fontWeight": 500,
          "pr": 3,
        }
      }
    >
      7:08
    </ForwardRef(TableCell)>
  </ForwardRef(TableRow)>
</ForwardRef(TableBody)>
`;
